saleor/dashboard/shipping/forms.py

```diff
@@ -183,6 +217,15 @@ def clean(self):
             )
         return data
 
+    def save(self, commit=True):
+        for k, v in self.cleaned_data.items():
+            if not isinstance(v, Money):
+                continue
+            if not v.currency:
+                v.currency = settings.DEFAULT_CURRENCY
```

Comment:
Couldn't this be handled by the `MoneyField` underneath?

Reply:
It should be but there is sometimes it just return `Money("1.23", "")`. I need to dig this, probably a bug on `django-prices`.

c.f. https://github.com/mirumee/saleor/pull/4639#discussion_r314691871